### PR TITLE
Instantiate App of Apps in argocd-helm module

### PR DESCRIPTION
**Pull request by raphink**

### CODE REVIEW

File: modules/argocd-helm/variables.tf [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  description = "Minio Access Key"
+  type        = string
+  default     = ""
+}
+
+variable "minio_secret_key" {
+  description = "Minio Secret Key"
+  type        = string
+  default     = ""
+}
```
Comment (mcanevet) [inline modules/argocd-helm/variables.tf:124]: We should group this 2 variables into a complexe one

File: modules/eks-aws/main.tf [hunk 2]
```diff
@@ -122,53 +122,23 @@ resource "aws_security_group_rule" "workers_ingress_healthcheck_http" {
...
-
-        oauth2_proxy_extra_args          = []
-        grafana_generic_oauth_extra_args = {}
-      }
-    ),
+  repo_url = var.repo_url
+  target_revision = var.target_revision
+  extra_apps = var.extra_apps
+  cluster_name = var.cluster_name
+  base_domain = var.base_domain
```
Comment (mcanevet) [inline modules/eks-aws/main.tf:129]: Pleas align `=` symbol

File: modules/eks-aws/main.tf [hunk 3]
```diff
@@ -122,53 +122,23 @@ resource "aws_security_group_rule" "workers_ingress_healthcheck_http" {
...
+
+  cluster_issuer                  = "letsencrypt-prod"
+  oidc_issuer_url                 = format("https://cognito-idp.%s.amazonaws.com/%s", data.aws_region.current.name, var.cognito_user_pool_id)
+  oauth2_oauth_url                = format("https://%s.auth.%s.amazoncognito.com/oauth2/authorize", var.cognito_user_pool_domain, data.aws_region.current.name)
+  oauth2_token_url                = format("https://%s.auth.%s.amazoncognito.com/oauth2/token", var.cognito_user_pool_domain, data.aws_region.current.name)
+  oauth2_api_url                  = format("https://%s.auth.%s.amazoncognito.com/oauth2/userInfo", var.cognito_user_pool_domain, data.aws_region.current.name)
+  client_id                       = aws_cognito_user_pool_client.client.id
+  client_secret                   = aws_cognito_user_pool_client.client.client_secret
+  loki_bucket_name                = aws_s3_bucket.loki.id,
+  enable_efs                      = var.enable_efs
```
Comment (mcanevet) [inline modules/eks-aws/main.tf:139]: Please remove extra spaces before `=` symbol

File: modules/k3os-libvirt/main.tf [hunk 4]
```diff
@@ -98,10 +80,11 @@ resource "helm_release" "app_of_apps" {
   ]
 
   depends_on = [
-    module.argocd,
+    module.cluster,
   ]
 }
 
+
```
Comment (mcanevet) [inline modules/k3os-libvirt/main.tf:87]: Please remove this

**Review by mcanevet — CHANGES_REQUESTED** [review]

**Comment by mcanevet** [comment]
I guess we should add a linting step into the pipeline...

**Comment by raphink** [comment]
I'm keeping the complex variables PR into a separate branch (https://github.com/raphink/camptocamp-devops-stack/tree/app_of_apps_complex_vars) so we can do that later

**Comment by mcanevet** [comment]
We need the same for aks-azure

**Comment by raphink** [comment]
Superceded by #351

**Pull request closed without merging**